ratis-logservice/src/main/java/org/apache/ratis/logservice/api/LogService.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ratis.logservice.api;
+
+import java.util.Iterator;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Entry point for interacting with the Ratis LogService.
+ */
+public interface LogService {
+
+  /**
+   * Creates a new {@link LogStream} identified by the given name. Throws
+   * an exception if a {@link LogStream} with the given name already exists.
+   *
+   * @param name Unique name for this LogStream.
+   */
+  CompletableFuture<LogStream> createLog(LogName name);
```

Comment:
While creating log , we may need to add some metadata for the stream, for eg:- Details of SERDE classes, compression etc.
And need the corresponding API in LogStream to get the MetaData.

Reply:
> we may need to add some metadata for the stream, for eg:- Details of SERDE classes, compression etc.

Good point. I can make a basic configuration class which just wraps a `Map<String,String>` for now? How does that sound while we flesh out what kind of metadata we'll have?